utils/src/main/kotlin/OrtAuthenticator.kt

```diff
@@ -74,15 +74,54 @@ class OrtAuthenticator(private val original: Authenticator? = null) : Authentica
     }
 
     override fun getPasswordAuthentication(): PasswordAuthentication? {
-        if (requestorType == RequestorType.PROXY) {
-            val proxySelector = ProxySelector.getDefault()
-            if (proxySelector is OrtProxySelector) {
-                val type = requestingProtocol.toProxyType() ?: return null
-                val proxy = Proxy(type, InetSocketAddress(requestingHost, requestingPort))
-                return proxySelector.getProxyAuthentication(proxy)
+        when (requestorType) {
+            RequestorType.PROXY -> {
+                val proxySelector = ProxySelector.getDefault()
+                if (proxySelector is OrtProxySelector) {
+                    val type = requestingProtocol.toProxyType() ?: return null
+                    val proxy = Proxy(type, InetSocketAddress(requestingHost, requestingPort))
+                    return proxySelector.getProxyAuthentication(proxy)
+                }
+            }
+
+            RequestorType.SERVER -> {
+                val netrcFile = getUserHomeDirectory().resolve(".netrc")
+                if (netrcFile.isFile) {
+                    getNetrcAuthentication(netrcFile.readText(), requestingHost)?.let { return it }
+                }
+
```

Comment:
Would it make sense to also log a warning if we reach this line?

Reply:
I don't think so. If at all maybe an `info`, but I believe even not doing that is fine.